x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/support/mapper/NativeRoleMappingStore.java

```diff
@@ -349,17 +351,16 @@ public void resolveRoles(UserData user, ActionListener<Set<String>> listener) {
         getRoleMappings(null, ActionListener.wrap(
                 mappings -> {
                     final ExpressionModel model = user.asModel();
-                    Stream<ExpressionRoleMapping> stream = mappings.stream()
-                            .filter(ExpressionRoleMapping::isEnabled)
-                            .filter(m -> m.getExpression().match(model));
-                    if (logger.isTraceEnabled()) {
-                        stream = stream.map(m -> {
-                            logger.trace("User [{}] matches role-mapping [{}] with roles [{}]", user.getUsername(), m.getName(),
-                                    m.getRoles());
-                            return m;
-                        });
-                    }
-                    final Set<String> roles = stream.flatMap(m -> m.getRoles().stream()).collect(Collectors.toSet());
+                    final Set<String> roles = mappings.stream()
+                        .filter(ExpressionRoleMapping::isEnabled)
+                        .filter(m -> m.getExpression().match(model))
+                        .flatMap(m -> {
```

Comment:
Why remove the `logger.isTraceEnabled()` guard?

Reply:
It wasn't strictly ever needed, and now that I need to flatMap over `m.getRoleNames` it seems cleaner to merge it into that block.